### ISSUE TITLE
POS Error when scanning items to a sale

### ISSUE THREAD

**Issue by josehjay** [frappe/erpnext#34684]
### Information about bug

I am getting a server error when I try scanning items to a sale in pos, this is after updating frappe to v14.20.0 and erpnext to v14.30.0 

### Module

other

### Version

Frappe version : 14.20.0
Erpnext version : 14.30.0

### Installation method

manual install

### Relevant log output / Stack trace / Full Error Message.

```shell
### App Versions

{
	"csf_ke": "2.1.5",
	"erpnext": "14.20.0",
	"frappe": "14.30.0",
	"hrms": "14.1.3",
	"payments": "0.0.1"
}
```
### Route
```
point-of-sale
```
### Traceback
```
Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 66, in application
    response = frappe.api.handle()
  File "apps/frappe/frappe/api.py", line 54, in handle
    return frappe.handler.handle()
  File "apps/frappe/frappe/handler.py", line 45, in handle
    data = execute_cmd(cmd)
  File "apps/frappe/frappe/handler.py", line 83, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "apps/frappe/frappe/__init__.py", line 1607, in call
    return fn(*args, **newargs)
  File "apps/erpnext/erpnext/selling/page/point_of_sale/point_of_sale.py", line 96, in get_items
    result = search_by_term(search_term, warehouse, price_list) or []
  File "apps/erpnext/erpnext/selling/page/point_of_sale/point_of_sale.py", line 52, in search_by_term
    item_stock_qty = item_stock_qty // item.get("conversion_factor")
TypeError: unsupported operand type(s) for //: 'float' and 'NoneType'

```
### Request Data
```
{
	"type": "POST",
	"args": {
		"start": 0,
		"page_length": 40,
		"price_list": "Standard Selling",
		"item_group": "All Item Groups",
		"search_term": "9780195749618",
		"pos_profile": "Terminal 2"
	},
	"freeze": true,
	"headers": {},
	"error_handlers": {},
	"url": "/api/method/erpnext.selling.page.point_of_sale.point_of_sale.get_items"
}
```
### Response Data
```
{
	"exception": "TypeError: unsupported operand type(s) for //: 'float' and 'NoneType'"
}
```
```

**Comment by FearsomePhobia** [comment]
Came across this issue too today, it appears to be related to UOM barcode scanning.

When no UOM is specified with the barcode (it should be using the item's default UOM) you get this exception:
`TypeError: unsupported operand type(s) for //: 'float' and 'NoneType'`

If a UOM is specified with the barcode you get this exception instead:
`pymysql.err.OperationalError: (1054, "Unknown column 'stock_uom' in 'field list'")`

Hope this issue gets fixed soon as POS barcode scanning is an extremely critical requirement for any retail store.

**Comment by ahmadpak** [comment]
I am facing the same issue.

**Comment by ravi854** [comment]
I am also facing the same issue

**Comment by ravi854** [comment]
For work around added custom link field stock_uom in Item Price doc type as it is looking for stock_uom field in Item Price.
It scanning the Item but UOM conversion not working on scanning

**Comment by salmenktata** [comment]
I am also facing the same issue do you tell us if there is any fix to this please ?

**Comment by ravi854** [comment]
I commented the line no 52  _"item_stock_qty = item_stock_qty // item.get("conversion_factor")"_ and added custom link field stock_uom in Item Price doc type.
its scanning the barcode with and without specifing UOM with barcode.

**Comment by xhijack** [comment]
Me too. Does anyone solve this bug?

**Comment by Nihantra-Patel** [comment]
It's fixed in above version: #35710

Kindly upgrade to version 14 or 15, then refresh and check again because lots of bugs fixed in the new version.

Everything is working fine. No problems were encountered in the latest versions 14 and 15.
